cmd/troubleshoot/cli/run.go

```diff
@@ -678,134 +372,50 @@ func getExpectedContentType(uploadURL string) string {
 	return parsedURL.Query().Get("Content-Type")
 }
 
-func callbackSupportBundleAPI(r *troubleshootv1beta2.ResultRequest, archivePath string) error {
-	for {
-		req, err := http.NewRequest(r.Method, r.URI, nil)
-		if err != nil {
-			return errors.Wrap(err, "create request")
-		}
-
-		resp, err := httpClient.Do(req)
-		if err != nil {
-			if shouldRetryRequest(err) {
-				continue
-			}
-			return errors.Wrap(err, "execute request")
-		}
-
-		if resp.StatusCode >= 300 {
-			return fmt.Errorf("unexpected status code %d", resp.StatusCode)
-		}
-
-		break
-	}
-	return nil
-}
-
-func tarSupportBundleDir(inputDir, outputFilename string) error {
-	fileWriter, err := os.Create(outputFilename)
-	if err != nil {
-		return errors.Wrap(err, "failed to create output file")
-	}
-	defer fileWriter.Close()
-
-	gzipWriter := gzip.NewWriter(fileWriter)
-	defer gzipWriter.Close()
-
-	tarWriter := tar.NewWriter(gzipWriter)
-	defer tarWriter.Close()
-
-	err = filepath.Walk(inputDir, func(filename string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-
-		fileMode := info.Mode()
-		if !fileMode.IsRegular() { // support bundle can have only files
-			return nil
-		}
-
-		parentDirName := filepath.Dir(inputDir) // this is to have the files inside a subdirectory
-		nameInArchive, err := filepath.Rel(parentDirName, filename)
-		if err != nil {
-			return errors.Wrap(err, "failed to create relative file name")
-		}
-
-		// tar.FileInfoHeader call causes a crash in static builds
-		// https://github.com/golang/go/issues/24787
-		hdr := &tar.Header{
-			Name:     nameInArchive,
-			ModTime:  info.ModTime(),
-			Mode:     int64(fileMode.Perm()),
-			Typeflag: tar.TypeReg,
-			Size:     info.Size(),
-		}
-
-		err = tarWriter.WriteHeader(hdr)
-		if err != nil {
-			return errors.Wrap(err, "failed to write tar header")
-		}
-
-		err = func() error {
-			fileReader, err := os.Open(filename)
-			if err != nil {
-				return errors.Wrap(err, "failed to open source file")
-			}
-			defer fileReader.Close()
-
-			_, err = io.Copy(tarWriter, fileReader)
-			if err != nil {
-				return errors.Wrap(err, "failed to copy file into archive")
-			}
-
-			return nil
-		}()
-		if err != nil {
-			return err
-		}
-
-		return nil
-	})
-	if err != nil {
-		return errors.Wrap(err, "failed to walk source dir")
-	}
-
-	return nil
-}
-
-type CollectorFailure struct {
-	Collector *troubleshootv1beta2.Collect
-	Failure   string
-}
-
```

Comment:
As far as I can tell this isn't used in troubleshoot or kots

Reply:
indeed